src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingPrisonOffenderEventListener.kt

```diff
@@ -36,10 +36,8 @@ class SentencingPrisonOffenderEventListener(
       if (eventFeatureSwitch.isEnabled(eventType)) when (eventType) {
         "SENTENCE_ADJUSTMENT_UPSERTED" -> sentencingSynchronisationService.synchroniseSentenceAdjustmentCreateOrUpdate((sqsMessage.Message.fromJson()))
         "SENTENCE_ADJUSTMENT_DELETED" -> sentencingSynchronisationService.synchroniseSentenceAdjustmentDelete((sqsMessage.Message.fromJson()))
-
-        "KEY_DATE_ADJUSTMENT_UPSERTED", "KEY_DATE_ADJUSTMENT_DELETED" -> {
-          log.debug("received $eventType Offender event but right now not doing anything with it")
-        }
+        "KEY_DATE_ADJUSTMENT_UPSERTED" -> sentencingSynchronisationService.synchroniseKeyDateAdjustmentCreateOrUpdate((sqsMessage.Message.fromJson()))
```

Comment:
Why upserts instead of create or update? When calling the API isn't PUT idempotent but POST non-idempotent?

Reply:
NOMIS triggers only emit a single event for creates and updates